nexus/client.go

```diff
@@ -0,0 +1,543 @@
+		err = errEmptyOperationName
+		return
+	}
+	var b []byte
+	b, err = json.Marshal(v)
+	if err != nil {
+		return
+	}
+	options.Operation = operation
+	options.Header = http.Header{headerContentType: []string{contentTypeJSON}}
+	options.Body = bytes.NewReader(b)
+	return
+}
+
+func (o *ExecuteOperationOptions) intoStartOptions() StartOperationOptions {
+	return StartOperationOptions{
+		Operation:   o.Operation,
+		CallbackURL: o.CallbackURL,
+		RequestID:   o.RequestID,
+		Header:      o.Header,
+		Body:        o.Body,
+	}
+}
+
+func (o *ExecuteOperationOptions) intoGetResultOptions() (options GetOperationResultOptions) {
+	options.Header = o.Header
+	if options.Header != nil {
+		options.Header = options.Header.Clone()
+		options.Header.Del(headerContentType)
+	}
+	if o.Wait <= 0 {
+		options.Wait = time.Duration(math.MaxInt64)
+	} else {
+		options.Wait = o.Wait
+	}
+	return
+}
+
+// ExecuteOperation is a helper for starting an operation and waiting for its completion.
+//
+// For asynchronous operations, the client will long poll for their result, issuing one or more requests until the
+// wait period provided via [ExecuteOperationOptions] exceeds, in which case an [ErrOperationStillRunning] error is
+// returned.
+//
+// The wait time is capped to the deadline of the provided context. Make sure to handle both context deadline errors and
+// [ErrOperationStillRunning].
+//
+// Note that the wait period is enforced by the server and may not be respected if the server is misbehaving. Set the
+// context deadline to the max allowed wait period to ensure this call returns in a timely fashion.
+//
+// ⚠️ If this method completes successfully, the returned response's body must be read in its entirety and closed to
+// free up the underlying connection.
+func (c *Client) ExecuteOperation(ctx context.Context, request ExecuteOperationOptions) (*http.Response, error) {
+	result, err := c.StartOperation(ctx, request.intoStartOptions())
+	if err != nil {
+		return nil, err
+	}
+	if result.Successful != nil {
+		return result.Successful, nil
+	}
+	handle := result.Pending
+	return handle.GetResult(ctx, request.intoGetResultOptions())
+}
+
+// NewHandle gets a handle to an asynchronous operation by name and ID.
+// Does not incur a trip to the server.
+// Fails if provided an invalid operation or ID.
+func (c *Client) NewHandle(operation string, operationID string) (*OperationHandle, error) {
+	var es []error
+	if operation == "" {
+		es = append(es, errEmptyOperationName)
+	}
+	if operationID == "" {
+		es = append(es, errEmptyOperationID)
+	}
+	if len(es) > 0 {
+		return nil, errors.Join(es...)
+	}
+	return &OperationHandle{
+		client:    c,
+		Operation: operation,
+		ID:        operationID,
+	}, nil
+}
+
+func (c *Client) sendGetOperationRequest(ctx context.Context, request *http.Request) (*http.Response, error) {
+	response, err := c.options.HTTPCaller(request)
+	if err != nil {
+		return nil, err
+	}
+
+	if response.StatusCode == http.StatusOK {
```

Comment:
:+1: On not also supporting other 2xx statuses here. This is a nice, simple thing for Nexus clients and users to know that a single status for this call is success